lib/shelley/test/data/smash/api/v1/metadata/ec28f33dcbe6d6400a1e5e339bd0647c0973ca6c0cf9c2bbe6838dc6/af7789093aa1fbff0f6e7026ecdbbd470f04e9a0415e1e7e6b9f1d54cd353126

```diff
@@ -0,0 +1 @@
+{"homepage":"https://iohk.io","name":"Genesis Pool A","ticker":"GPA","description":null}
```

Comment:
I'd be better to not duplicate test data here and rely on one single source of truth. Metadata for pools are declared in: 

https://github.com/input-output-hk/cardano-wallet/blob/0fb513921ff14cfc53aa3d508b170c4d2c24c878/lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs#L1402-L1513

Metadata are written to disk during the test cluster setup for each stake pool when the corresponding stake pool is registered, and a static server is started to serve them in the "direct" mode. 

https://github.com/input-output-hk/cardano-wallet/blob/0fb513921ff14cfc53aa3d508b170c4d2c24c878/lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs#L788-L820

We should use a similar strategy for SMASH. It's okay to read the entire MVar upfront and write all metadata in temporary directory for the SMASH stub.

Reply:
I suggest that the files be the source of truth then, not the haskell code. That also avoids hide and seek with metadata hash, which now depends on the aeson encoding.

Reply:
> That also avoids hide and seek with metadata hash, which currently depends on the aeson encoding.

This is why we use code for this, so it is easily reproducible and does not depend on someone's editor behavior. Please stick to the current pattern and use the Haskell code as source to generate files.